examples/evm_bots.py

```diff
@@ -512,78 +542,84 @@ def log_and_show_block_info():
+    return hyperdrive
 
 
 if __name__ == "__main__":
-    config = get_config()  # Instantiate the config using the command line arguments as overrides.
-    output_utils.setup_logging(log_filename=config.log_filename, log_level=config.log_level)
-
-    deployer = None  # pylint: disable=invalid-name
+    parsed_args = get_argparser().parse_args()
+    experiment_config = get_config(parsed_args)  # Instantiate the config using the command line arguments as overrides.
+    output_utils.setup_logging(log_filename=experiment_config.log_filename, log_level=experiment_config.log_level)
+    account_deployer = None  # pylint: disable=invalid-name
     # Set up ape
-    if config.scratch["devnet"]:  # if devnet setting is enabled
-        simulator = get_simulator(config)  # Instantiate the sim market
-        k, ps = "ethereum:local:foundry", {"fork_url": "http://localhost:8547", "port": 8549}
-        provider: ProviderAPI = ape.networks.parse_network_choice(k, provider_settings=ps).push_provider()
-        deployer = ape.accounts.test_accounts[0]
-        deployer.balance += int(1e18)  # eth, for spending on gas, not erc20
+    if experiment_config.scratch["devnet"]:  # if devnet setting is enabled
+        simulator = get_simulator(experiment_config)  # Instantiate the sim market
+        ps = {"port": 8549}
+        provider: ProviderAPI = ape.networks.parse_network_choice(
+            "ethereum:local:foundry", provider_settings=ps
+        ).push_provider()
+        account_deployer = ape.accounts.test_accounts[0]
+        account_deployer.balance += int(1e18)  # eth, for spending on gas, not erc20
         print(f"connected to devnet fork, latest block {provider.get_block('latest').number:,.0f}")
     else:
         provider: ProviderAPI = ape.networks.parse_network_choice(f"ethereum:{PROVIDER_STRING}").push_provider()
     project = ape_utils.HyperdriveProject(Path.cwd())
-    base: ContractInstance
-    if config.scratch["devnet"]:
-        assert isinstance(deployer, TestAccountAPI)
-        base: ContractInstance = deployer.deploy(project.get_contract("ERC20Mintable"))
-        fixed_math: ContractInstance = deployer.deploy(project.get_contract("MockFixedPointMath"))
+    base_instance: ContractInstance = None
+    if experiment_config.scratch["devnet"]:
+        assert isinstance(account_deployer, TestAccountAPI)
+        project_container: ContractContainer = project.get_contract("ERC20Mintable")
+        fixed_math: ContractInstance = account_deployer.deploy(project.get_contract("MockFixedPointMath"))
+        base_instance = account_deployer.deploy(project_container)
     else:
-        base: ContractInstance = ape_utils.get_instance(DAI_ADDRESS, provider=provider)  # sDai
-    if config.scratch["devnet"]:
-        hyperdrive: ContractInstance = deploy_hyperdrive()
+        base_instance = ape_utils.get_instance(DAI_ADDRESS, provider=provider)  # sDai
+    hyperdrive_instance: ContractInstance = None
```

Comment:
no need for this line, since `hyperdrive_instance` is defined on both conditional paths. unless you were just looking to add typehints (which you could do on each of the 2 conditional paths... or once here).

Reply:
Yeah I did that on purpose, it's the same I just like this flow better.